src/js/view/controls/components/menu/menu.js

```diff
@@ -326,76 +304,67 @@ export default class Menu extends Events {
         }
         delete this.children[name];
         menu.destroy();
-        this.trigger('menuRemoved', name);
+        this.mainMenu.trigger('menuRemoved', name);
     }
     open(evt) {
         const mainMenuVisible = this.mainMenu.visible;
         let focusEl;
-        if (this.isSubmenu) {
-            if (!this.items.length) {
-                return;
-            }
-            const sourceEvent = evt && evt.sourceEvent;
-            const firstItem = this.topbar ? this.topbar.firstChild : this.items[0].el;
-            const lastItem = this.items[this.items.length - 1].el;
-            const isKeydown = sourceEvent && sourceEvent.type === 'keydown';
-            const key = isKeydown && normalizeKey(sourceEvent.key);
-            const itemTarget = key === 'Up' ? lastItem : firstItem;
-            if (this.visible && !this.openMenus.length) {
-                if (this.items.length && isKeydown) {
-                    itemTarget.focus();
-                }
-                return;
-            }
-            const { mainMenu, parentMenu, categoryButton } = this;
-            parentMenu.openMenus.push(this.name);
-            if (parentMenu.openMenus.length > 1) {
-                parentMenu.closeChildren(this.name);
-            }
-            if (categoryButton) {
-                categoryButton.element().setAttribute('aria-checked', 'true');
-            }
-            if (parentMenu.isSubmenu) {
-                parentMenu.el.classList.remove('jw-settings-submenu-active');
-                mainMenu.topbar.classList.add('jw-nested-menu-open');
-                const menuTitle = mainMenu.topbar.querySelector('.jw-settings-topbar-text');
-                menuTitle.setAttribute('name', this.name);
-                menuTitle.innerText = this.title || this.name;
-                mainMenu.backButton.show();
-                backButtonTarget = this.parentMenu;
-                focusEl = menuTitle;
-            } else {
-                mainMenu.topbar.classList.remove('jw-nested-menu-open');
-                if (mainMenu.backButton) {
-                    mainMenu.backButton.hide();
-                }
-            }
-            this.el.classList.add('jw-settings-submenu-active');
-            if (mainMenuVisible && isKeydown) {
-                focusEl = itemTarget;
-            } else if (!mainMenuVisible) {
-                mainMenu.open(evt);
-                focusEl = categoryButton.element();
-                if (categoryButton && categoryButton.tooltip && !isKeydown) {
-                    categoryButton.tooltip.suppress = true;
-                        
-                }
-            }
-            if (this.openMenus.length) {
-                this.closeChildren();
-            }
-            if (focusEl) {
-                focusEl.focus();
+        if (!this.items.length) {
```

Comment:
the only difference in this code block appears to be the `isSubmenu` check being removed, is that correct?

Reply:
Yep, we overload open in mainMenu, so we don't need to run a detection check. Also, removed the else block for same reason